lib/eosknowledgeprivate/ekn-web-view-dialog.c

```diff
@@ -0,0 +1,239 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/*
+ * ekn-web-view-dialog.c
+ *
+ * Copyright (C) 2016 Endless Mobile, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * Author: Juan Pablo Ugarte <ugarte@endlessm.com>
+ *
+ */
+
+#include "ekn-web-view-dialog.h"
+
+struct EknWebViewDialogPrivate
+{
+  gchar *home_uri;
+
+  WebKitWebView *webview;
+  GtkHeaderBar  *headerbar;
+  GtkWidget     *back;
+  GtkWidget     *forward;
+};
+
+enum
+{
+  PROP_0,
+
+  PROP_WEBVIEW,
+  PROP_HOME_URI
+};
+
+G_DEFINE_TYPE_WITH_PRIVATE (EknWebViewDialog,
+                            ekn_web_view_dialog,
+                            GTK_TYPE_WINDOW);
+
+static void
+on_back_forward_list_changed (WebKitBackForwardList     *list,
+                              WebKitBackForwardListItem *items_added,
+                              GList                     *items_removed,
+                              EknWebViewDialog          *dialog)
+{
+  EknWebViewDialogPrivate *priv = dialog->priv;
+
+  gtk_widget_set_sensitive (priv->back, webkit_web_view_can_go_back (priv->webview));
+  gtk_widget_set_sensitive (priv->forward, webkit_web_view_can_go_forward (priv->webview));
+}
+
+static void
+ekn_web_view_dialog_init (EknWebViewDialog *dialog)
+{
+  EknWebViewDialogPrivate *priv;
+
+  dialog->priv = priv = ekn_web_view_dialog_get_instance_private (dialog);
```

Comment:
We tend not to do this in other code, just use `...get_instance_private()` everywhere since that is not really more than a pointer access either.